GTG/gtk/general_preferences.py

```diff
@@ -0,0 +1,249 @@
+# -----------------------------------------------------------------------------
+
+"""General Preferences represents one of the children
+in Preferences window. It is viewed in its stack.
+It enables user to set the most general settings of GTG."""
+
+import os
+
+from gi.repository import Gtk, Gdk
+
+from GTG.core.dirs import UI_DIR
+from GTG.core.translations import _
+from GTG.tools import autostart
+from GTG.tools import shortcut
+
+
+class GeneralPreferences(object):
+
+    GENERAL_PREFERENCES_UI = os.path.join(UI_DIR, "general_preferences.ui")
+    INVALID_COLOR = Gdk.Color(50000, 0, 0)
+
+    def __init__(self, req, vmanager):
+        self.req = req
+        self.config = self.req.get_config('browser')
+        builder = Gtk.Builder()
+        builder.add_from_file(self.GENERAL_PREFERENCES_UI)
+
+        self.ui_widget = builder.get_object("general_pref_window")
+        self.autostart_button = builder.get_object("autostart_button")
+        self.preview_button = builder.get_object("preview_button")
+        self.bg_color_button = builder.get_object("bg_color_button")
+        self.shortcut_button = builder.get_object("shortcut_button")
+        self.font_button = builder.get_object("font_button")
+        self.shortcut_popover = builder.get_object("shortcut_popover")
+        self.set_shortcut = builder.get_object("set_shortcut")
+
+        self.shortcut = ShortcutWidget(builder)
+
+        self.timer = vmanager.timer
+        self.refresh_time = builder.get_object("time_entry")
+
+        self.set_editor_font()
+
+        builder.connect_signals(self)
+
+    def get_ui(self):
+        '''
+        This method returns widget displayed in Preferences window.
+        '''
+        return self.ui_widget
+
+    def on_shortcut_popover(self, widget):
+        self.shortcut_popover.show()
+
+    def set_editor_font(self):
+        editor_font = self.config.get("font_name")
+        if editor_font == "":
+            font = self.ui_widget.get_style_context().get_font(
+                Gtk.StateFlags.NORMAL)
+            editor_font = font.to_string()
+        self.font_button.set_font_name(editor_font)
+
+    def _refresh_preferences_store(self):
+        """ Sets the correct value in the preferences checkboxes """
+        self.autostart_button.set_active(autostart.is_enabled())
+
+        self.shortcut.refresh_accel()
+
+        show_preview = self.config.get("contents_preview_enable")
+        self.preview_button.set_active(show_preview)
+
+        bg_color = self.config.get("bg_color_enable")
+        self.bg_color_button.set_active(bg_color)
+
+        self.refresh_time.set_text(self.timer.get_formatted_time())
+        self.refresh_time.modify_fg(Gtk.StateFlags.NORMAL, None)
+
+    def _refresh_task_browser(self):
+        """ Refresh tasks in task browser """
+        task_tree = self.req.get_tasks_tree(refresh=False).get_basetree()
+        task_tree.refresh_all()
+
+    def on_valid_time_check(self, widget):
```

Comment:
Can you please provide docstring here? It might not be immediately obvious to new contributors what's the difference between `on_valid_time_check` and `on_leave_time_entry`